### DISCUSSION TITLE
Custom Module Without EF?

### DISCUSSION THREAD

**Discussion by vnomus** [oqtane/oqtane.framework#3345] [Q&A] [answered]
Are there any examples of custom modules that don't utilize EF. I need to create a module that stores it's data in MongoDB. It looks like the Module Creator assumes a relational data store, and as I'm new to Razor, Blazor, Oqtane, and EF, I'm having difficulty trying to weed out what's required in this scenario..... Any and all help much appreciated!....

**Comment by sbwalker** [comment, marked as answer]
@vnomus  Here is a blog on how to use Dapper for data access and SQL Scripts for migrations:

https://www.oqtane.org/blog/!/67/dapper-sql-scripts

Hopefully this is helpful

**Reply by vnomus** [reply]
Thanks Shawn!....
